### PR TITLE
Disable test added in 1611 that contains data race

### PR DESCRIPTION
**Pull request by pmorie**

### CODE REVIEW

**Comment by pmorie** [comment]
Merging to fix the build

cc @jpeeler @staebler

**Comment by staebler** [comment]
You can read my comment in the original PR for why this test is flaky.

https://github.com/kubernetes-incubator/service-catalog/pull/1611#discussion_r157276274

File: test/integration/controller_instance_test.go [hunk 1]
```diff
@@ -1261,43 +1261,43 @@ func TestDeleteServiceInstance(t *testing.T) {
...
-			testFunction: func(ct *controllerTest) {
-				verifyCondition := v1beta1.ServiceInstanceCondition{
-					Type:   v1beta1.ServiceInstanceConditionReady,
-					Status: v1beta1.ConditionFalse,
-					Reason: "ProvisionRequestInFlight",
-				}
-				if err := util.WaitForInstanceCondition(ct.client, testNamespace, testInstanceName, verifyCondition); err != nil {
-					t.Fatalf("error waiting for instance condition: %v", err)
-				}
-				ct.osbClient.PollLastOperationReaction = &fakeosb.PollLastOperationReaction{
```
Comment (staebler) [inline test/integration/controller_instance_test.go]: This `PollLastOperationReaction` being accessed by the thread that is handling the PollLastOperation request in the fake broker. At the same time, it is being modified in the thread running the test.

Comment (staebler) [reply test/integration/controller_instance_test.go]: This is precisely why the `DynamicPollLastOperationReaction` and its ilk were adding to the fake broker. The dynamic reactions can use atomic variables that are controlled entirely by the test to change the behavior of the reaction without causing data races.

**Pull request merged**